### PR TITLE
Tutorial: "How to make visualizations match filters"

### PR DESCRIPTION
**Pull request by meganelacheny**

### CODE REVIEW

File: ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+
+
+.. role:: tutorial-keycap(emphasis)
+
+
+.. rst-class:: header-information-tutorial
+
+    ★★★ Advanced - time: 45 minutes
+
+*This tutorial will teach you how to create different visualizations options to match related filters. This can be useful especially with charts, since they can sometimes become useless once related filters are activated. This tutorial will be fully customizable and reusable in any dashboard or piece of content created with HTML/CSS on OpenDataSoft.*
```
Comment (juliendbg) [inline ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst]: `visualization options`

File: ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+    </div>
+ </div>
+
+.. rst-class:: img-hide
+
+    SCREENSHOT
+
+.. rst-class:: block-step
+
+    2 + Before going any further, declare the unique context of your page, using the `odsDatasetContext <https://help.opendatasoft.com/widgets/#/api/ods-widgets.directive:odsDatasetContext>`_ widget. This will allow our widgets to interact with one another -and it always is the proper way to build a dashboard.
```
Comment (juliendbg) [inline ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst]: there is a space missing between the dash and "and"

File: ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+
+ </ods-dataset-context>
+
+.. rst-class:: img-hide
+
+    SCREENSHOT
+
+.. rst-class:: block-step
+
+    4 + Click on :tutorial-keycap:`Preview`. Technically, everything works well: the chart indeed updates according to the chosen filter option. However, the filtered chart becomes completely useless, because it only displays one only point.
```
Comment (juliendbg) [inline ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst]: `it only displays one point` or `it only displays one single point`

File: ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst [hunk 4]
```diff
@@ -0,0 +1,529 @@
...
+
+.. rst-class:: img-hide
+
+    SCREENSHOT
+
+We need a second option visualization, that will only be displayed in case the filter is used. This second option visualization will replace the default, unfiltered chart that becomes useless once filtered. Let's do this!
```
Comment (juliendbg) [inline ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst]: - `we need a second visualization option`? (twice here + more later), also, I don't really understand what it means. What about "alternative visualization" or "specific visualization"?
- `that will only be displayed if the filter is used`?

File: ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst [hunk 5]
```diff
@@ -0,0 +1,529 @@
...
+
+ </ods-dataset-context>
+
+.. rst-class:: img-hide
+
+    SCREENSHOT
+
+.. rst-class:: block-step
+
+    6 + Now is the time to tackle AngularJS! We will need to add an AngularJS directive to both visualizations: the default one (that should NOT be displayed if the filter is used) and the 2nd option chart (that should ONLY be displayed if the filter is used). Since any AngularJS directive must be added into an HTML tag, and seeing our code, the best way to do in order to keep our code clean is to add ``div`` tags. Wrap each chart between ``div`` tags.
```
Comment (juliendbg) [inline ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst]: "Now it is ~the~ time"

File: ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst [hunk 6]
```diff
@@ -0,0 +1,529 @@
...
+
+- ``refine`` indicates that we use a filter parameter.
+- ``.field_ID`` allows us to precise which specific filter we are going to use. ``field_ID`` must be replaced by the technical identifier of the field that is used as filter.
+
+Basically, the ``ng-if`` syntax could be read as such: "If ``mycontext`` is filtered with ``field_ID``, [display the related HTML elements]".
+
+.. rst-class:: block-step
+
+    8 + Add the ``ng-if`` directive into the ``div`` tag wrapping the 2nd option chart. Don't forget to replace ``mycontext`` by the name of page context (see step 2) and ``field_ID`` by the technical identifier of the dataset field that we use as filter (see step 3). In this tutorial, ``mycontext`` is replaced by ``worlduniversityranking`` and ``field_ID`` is replaced by ``country``.
```
Comment (juliendbg) [inline ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst]: `to replace ``mycontext`` *with* the name`

File: ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst [hunk 7]
```diff
@@ -0,0 +1,529 @@
...
+    SCREENSHOT
+
+
+
+
+.. rst-class:: block-prerequisite-congratulation
+
+ **CONGRATULATIONS!**
+
+ You have finished this tutorial and you now know how to create different visualizations options to make sure they match the filters of the page. And hopefully your now feel comfortable enough with the combination of ``ng-if`` and ``context.parameters`` to be able to use it again on your own!
```
Comment (juliendbg) [inline ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst]: `visualization options`

File: ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst [hunk 8]
```diff
@@ -0,0 +1,529 @@
...
+
+.. rst-class:: title-level-2
+
+    Create the base
+
+In this tutorial, we will create a page comprised of a visualization and its related dataset filter, using the `Shanghai World University Ranking <https://data.opendatasoft.com/explore/dataset/shanghai-world-university-ranking%40public/>`_ dataset that can be found on our OpenDataSoft data network.
+
+.. rst-class:: block-step
+
+    1 + Let's start our page by adding the widget code of the visualization of our choice. Here, we chose to display a scatter plot chart representing the top 20 of the best universities in the ranking. For each university, the chart shows their average score.
```
Comment (fpassaniti) [inline ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst]: Top 20 des pays ! pas des université !
tu fais déjà du top 10 coutry avec l'autre dataviz, et ton chart est en field-x = country

File: ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst [hunk 9]
```diff
@@ -0,0 +1,529 @@
...
+     <ods-chart align-month="true">
+        <ods-chart-query context="worlduniversityranking" field-x="country" maxpoints="20" sort="serie1-1">
+            <ods-chart-serie expression-y="total_score" chart-type="scatter" function-y="AVG" color="#FF515A" scientific-display="true">
+            </ods-chart-serie>
+        </ods-chart-query>
+     </ods-chart>
+     </div>
+
+    <div ng-if="worlduniversityranking.parameters['refine.country']">
+    <ods-chart align-month="true">
```
Comment (fpassaniti) [inline ods-tutorial/source/dashboard_modules/how_to_hide_or_show_widgets.rst]: petit détail : indentation manquante ici

**Pull request merged**